refactor: improve installation error message

sgoral-splunk opened the pull request:
**Issue number:** [ADDON-78105](https://splunk.atlassian.net/browse/ADDON-78105)
### PR Type

**What kind of change does this PR introduce?**
* [ ] Feature
* [ ] Bug Fix
* [x] Refactoring (no functional or API changes)
* [ ] Documentation Update
* [ ] Maintenance (dependency updates, CI, etc.)

## Summary

Improve error message for library installation process.

### Changes

Added error message log to `_pip_install` method.

### User experience

Better understanding why `ucc-gen` build failed due to `CouldNotInstallRequirements` error.

## Checklist

If an item doesn't apply to your changes, leave it unchecked.

* [x] I have performed a self-review of this change according to the [development guidelines](https://splunk.github.io/addonfactory-ucc-generator/contributing/#development-guidelines)
* [x] Tests have been added/modified to cover the changes [(testing doc)](https://splunk.github.io/addonfactory-ucc-generator/contributing/#build-and-test)
* [ ] Changes are documented
* [ ] PR title and description follows the [contributing principles](https://splunk.github.io/addonfactory-ucc-generator/contributing/#pull-requests)

kkedziak-splunk reviewed (dismissed):
LGTM 👍 If we now have the error message, do we still need the traceback? Also, what about stdout?

```
INFO: Installing requirements from ../splunk-example-ta/package/lib/requirements.txt
INFO: Executing: python3 -m pip install --upgrade pip
INFO: Executing: python3 -m pip install -r "../splunk-example-ta/package/lib/requirements.txt" --target "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/output/Splunk_TA_Example/lib" --no-compile --prefer-binary --ignore-installed
ERROR: Command (pip install) returned 1 status code
ERROR: Command execution failed with error message: ERROR: Could not find a version that satisfies the requirement somefakerequirementsdasdqw (from versions: none)
ERROR: No matching distribution found for somefakerequirementsdasdqw

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/main.py", line 244, in main
    build.generate(
  File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/commands/build.py", line 494, in generate
    install_python_libraries(
  File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 190, in install_python_libraries
    install_libraries(
  File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 276, in install_libraries
    _pip_install(
  File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 88, in _pip_install
    raise CouldNotInstallRequirements
splunk_add_on_ucc_framework.install_python_libraries.CouldNotInstallRequirements
```

artemrys commented:
@sgoral-splunk does it fix https://github.com/splunk/addonfactory-ucc-generator/issues/1599 as well?

artemrys commented on splunk_add_on_ucc_framework/install_python_libraries.py:
```diff
@@ -79,8 +79,12 @@ def _subprocess_run(
     cmd = f"{installer} -m pip install {command}"
     try:
-        return_code = _subprocess_run(command=cmd, command_desc=command_desc).returncode
+        subprocess_result = _subprocess_run(command=cmd, command_desc=command_desc)
+        return_code = subprocess_result.returncode
         if return_code != 0:
+            logger.error(
+                f"Command execution failed with error message: {subprocess_result.stderr.decode()}"
+            )
```
Should we break the whole `ucc-gen build` if this fails?

soleksy-splunk replied:
I believe the raise after that is breaking build. 

I personally would expect it fail if any of libs is not correctly installed.
Is there any benefit to not break it?

Theoretically after it fails (and build continues) someone might want/know how to add an library separately, but it seems to fishy.

But at the same time they can remove this lib from requirements and add them after.

Not sure if there is any other benefit/use case of build going further.

sgoral-splunk replied:
yes, the raise will break the build because in the `install_libraries` we do not handling this exception.

artemrys commented:
> > @sgoral-splunk does it fix #1599 as well?
> 
> it should, it's very similar to what was proposed [here](https://github.com/splunk/addonfactory-ucc-generator/issues/1599) but as a logger.error not exception msg.

I just attempted to install a random library and got 

```
...
ERROR: Command (pip install) returned 1 status code
ERROR: Command execution failed with error message: ERROR: Could not find a version that satisfies the requirement randomthing (from versions: none)
ERROR: No matching distribution found for randomthing

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/<user>/addonfactory-ucc-generator/splunk_add_on_ucc_framework/main.py", line 244, in main
    build.generate(
  File "/Users/<user>/addonfactory-ucc-generator/splunk_add_on_ucc_framework/commands/build.py", line 494, in generate
    install_python_libraries(
  File "/Users/<user>/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 190, in install_python_libraries
    install_libraries(
  File "/Users/<user>/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 276, in install_libraries
    _pip_install(
  File "/Users/<user>/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 88, in _pip_install
    raise CouldNotInstallRequirements
splunk_add_on_ucc_framework.install_python_libraries.CouldNotInstallRequirements
```

I think we can omit logging the complete exception in this case, what do you think?

sgoral-splunk commented:
> LGTM 👍 If we now have the error message, do we still need the traceback? Also, what about stdout?
> 
> ```
> INFO: Installing requirements from ../splunk-example-ta/package/lib/requirements.txt
> INFO: Executing: python3 -m pip install --upgrade pip
> INFO: Executing: python3 -m pip install -r "../splunk-example-ta/package/lib/requirements.txt" --target "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/output/Splunk_TA_Example/lib" --no-compile --prefer-binary --ignore-installed
> ERROR: Command (pip install) returned 1 status code
> ERROR: Command execution failed with error message: ERROR: Could not find a version that satisfies the requirement somefakerequirementsdasdqw (from versions: none)
> ERROR: No matching distribution found for somefakerequirementsdasdqw
> 
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/main.py", line 244, in main
>     build.generate(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/commands/build.py", line 494, in generate
>     install_python_libraries(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 190, in install_python_libraries
>     install_libraries(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 276, in install_libraries
>     _pip_install(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 88, in _pip_install
>     raise CouldNotInstallRequirements
> splunk_add_on_ucc_framework.install_python_libraries.CouldNotInstallRequirements
> ```

hmm you are right. I think in general traceback is not very useful for the user so we can remove it. In case of stdout i checked and there is no relevant information and in my opinion it would generate spam in the console.

sgoral-splunk commented:
> LGTM 👍 If we now have the error message, do we still need the traceback? Also, what about stdout?
> 
> ```
> INFO: Installing requirements from ../splunk-example-ta/package/lib/requirements.txt
> INFO: Executing: python3 -m pip install --upgrade pip
> INFO: Executing: python3 -m pip install -r "../splunk-example-ta/package/lib/requirements.txt" --target "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/output/Splunk_TA_Example/lib" --no-compile --prefer-binary --ignore-installed
> ERROR: Command (pip install) returned 1 status code
> ERROR: Command execution failed with error message: ERROR: Could not find a version that satisfies the requirement somefakerequirementsdasdqw (from versions: none)
> ERROR: No matching distribution found for somefakerequirementsdasdqw
> 
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/main.py", line 244, in main
>     build.generate(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/commands/build.py", line 494, in generate
>     install_python_libraries(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 190, in install_python_libraries
>     install_libraries(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 276, in install_libraries
>     _pip_install(
>   File "/Users/kkedziak/PycharmProjects/addonfactory-ucc-generator/splunk_add_on_ucc_framework/install_python_libraries.py", line 88, in _pip_install
>     raise CouldNotInstallRequirements
> splunk_add_on_ucc_framework.install_python_libraries.CouldNotInstallRequirements
> ```

I removed traceback and added try except clause to the install_command function
